doc/wol/Wake-on-LAN-HLD.md

```diff
@@ -0,0 +1,136 @@
+# Wake-on-LAN in SONiC
+
+## Table of Content
+
+- [Overview](#overview)
+- [Background](#background)
+- [Magic Packet](#magic-packet)
+- [CLI Design](#cli-design)
+- [gNOI Design](#gnoi-design)
+- [Test Plan](#test-plan)
+- [Reference](#reference)
+
+## Revision
+
+| Revision | Date       | Author     | Change Description |
+| -------- | ---------- | ---------- | ------------------ |
+| 1.0      | Nov 7 2023 | Zhijian Li | Initial proposal   |
+
+## Definitions/Abbreviations 
+
+| Abbreviation | Definition                                        |
+|--------------|---------------------------------------------------|
+| WoL          | **W**ake-**o**n-**L**AN                           |
+| gNOI         | **g**RPC **N**etwork **O**perations **I**nterface |
+| NIC          | **N**etwork **I**nterface **C**ontroller          |
+
+## Overview
+
+**Wake-on-LAN** (**WoL** or **WOL**) is an Ethernet or Token Ring computer networking standard that allows a computer to be turned on or awakened from sleep mode by a network message[^1]. This document describes the Wake-on-LAN feature design in SONiC.
+
+## Background
+
+Below diagram describes a common usage of WoL on SONiC switch, this HLD will focus on the green part:
+
+![WoL Background](./wol-background.png)
```

Comment:
Please expand `SONIC Switch` in the diagram and explain the components involved in generating the `Magic Packet`

Reply:
Thanks, I added a `Component` section with some pictures to illustrate that.